pkg/registry/apis/provisioning/repository/go-git/wrapper.go

```diff
@@ -363,11 +388,16 @@ func (g *GoGitRepo) maybeCommit(ctx context.Context, message string) error {
 
 // Delete implements repository.Repository.
 func (g *GoGitRepo) Delete(ctx context.Context, fpath string, ref string, message string) error {
-	fpath = safepath.Join(g.config.Spec.GitHub.Path, fpath)
 	if err := verifyPathWithoutRef(fpath, ref); err != nil {
 		return err
 	}
+
+	fpath = safepath.Join(g.config.Spec.GitHub.Path, fpath)
 	if _, err := g.tree.Remove(fpath); err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
```

Comment:
This is an error that all the repository implementations should return if the file doesn't exist. We probably have inconsistencies across them.